diff --git a/Source/WebCore/platform/graphics/OpenGLShims.h b/Source/WebCore/platform/graphics/OpenGLShims.h
index 7b194b2a6c26..f0feb6bb1886 100644
--- a/Source/WebCore/platform/graphics/OpenGLShims.h
+++ b/Source/WebCore/platform/graphics/OpenGLShims.h
@@ -20,6 +20,8 @@
 #ifndef OpenGLShims_h
 #define OpenGLShims_h
 
+#define GL_GLEXT_PROTOTYPES 1
+
 #if OS(WINDOWS)
 #include <GL/gl.h>
 #include <GLES2/gl2.h>
