diff -up webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h.sparc webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h
--- webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h.sparc	2009-10-01 15:06:02.000000000 -0400
+++ webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h	2010-03-22 13:27:50.011061289 -0400
@@ -301,8 +301,12 @@
 #if defined(__sparc__) && defined(__arch64__) || defined (__sparcv9)
 #define WTF_PLATFORM_SPARC64 1
 #define WTF_PLATFORM_BIG_ENDIAN 1
+#elif defined(__sparc__) /*PLATFORM(SPARC) */
+#define WTF_PLATFORM_SPARC 1
+#define WTF_PLATFORM_BIG_ENDIAN 1
 #endif
 
+
 /* PLATFORM(WINCE) && PLATFORM(QT)
    We can not determine the endianess at compile time. For
    Qt for Windows CE the endianess is specified in the
@@ -630,7 +634,7 @@
 #endif
 
 #if !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32) && !defined(WTF_USE_JSVALUE32_64)
-#if PLATFORM(X86_64) && (PLATFORM(DARWIN) || PLATFORM(LINUX))
+#if PLATFORM(X86_64) && (PLATFORM(DARWIN) || PLATFORM(LINUX)) || PLATFORM(SPARC64)
 #define WTF_USE_JSVALUE64 1
 #elif PLATFORM(ARM) || PLATFORM(PPC64)
 #define WTF_USE_JSVALUE32 1
