diff -up ./JavaScriptCore/wtf/Platform.h.old ./JavaScriptCore/wtf/Platform.h
--- ./JavaScriptCore/wtf/Platform.h.old	2009-08-22 15:00:39.306173267 -0700
+++ ./JavaScriptCore/wtf/Platform.h	2009-08-22 15:03:40.551182842 -0700
@@ -675,7 +675,7 @@
 #endif
 /* Setting this flag prevents the assembler from using RWX memory; this may improve
    security but currectly comes at a significant performance cost. */
-#if PLATFORM(ARM)
+#if PLATFORM(ARM) || PLATFORM(X86) || PLATFORM(X86_64)
 #define ENABLE_ASSEMBLER_WX_EXCLUSIVE 1
 #else
 #define ENABLE_ASSEMBLER_WX_EXCLUSIVE 0
