diff -up webkit-1.1.13/JavaScriptCore/wtf/Platform.h.no-execmem webkit-1.1.13/JavaScriptCore/wtf/Platform.h
--- webkit-1.1.13/JavaScriptCore/wtf/Platform.h.no-execmem	2009-08-25 09:27:08.517026703 -0400
+++ webkit-1.1.13/JavaScriptCore/wtf/Platform.h	2009-08-25 09:42:35.222265742 -0400
@@ -724,7 +724,7 @@
 #endif
 /* Setting this flag prevents the assembler from using RWX memory; this may improve
    security but currectly comes at a significant performance cost. */
-#if PLATFORM(IPHONE)
+#if PLATFORM(IPHONE) || PLATFORM(X86) || PLATFORM(X86_64)
 #define ENABLE_ASSEMBLER_WX_EXCLUSIVE 1
 #else
 #define ENABLE_ASSEMBLER_WX_EXCLUSIVE 0
