Index: trunk/Source/WebKit/gtk/webkit/webkitwebview.cpp
===================================================================
--- trunk/Source/WebKit/gtk/webkit/webkitwebview.cpp	(revision 113147)
+++ trunk/Source/WebKit/gtk/webkit/webkitwebview.cpp	(revision 114385)
@@ -3623,5 +3623,4 @@
 
     gtk_widget_set_can_focus(GTK_WIDGET(webView), TRUE);
-    gtk_widget_set_double_buffered(GTK_WIDGET(webView), FALSE);
 
     priv->mainFrame = WEBKIT_WEB_FRAME(webkit_web_frame_new(webView));
