diff -up webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h.s390 webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h
--- webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h.s390	2010-03-31 15:56:53.000000000 +0200
+++ webkit-1.1.15.4/JavaScriptCore/wtf/Platform.h	2010-03-31 15:57:32.000000000 +0200
@@ -306,6 +306,14 @@
 #define WTF_PLATFORM_BIG_ENDIAN 1
 #endif
 
+/* PLATFORM(S390*) */
+#if defined(__s390x__)
+#define WTF_PLATFORM_S390X 1
+#define WTF_PLATFORM_BIG_ENDIAN 1
+#elif defined(__s390__) /*PLATFORM(S390) */
+#define WTF_PLATFORM_S390 1
+#define WTF_PLATFORM_BIG_ENDIAN 1
+#endif
 
 /* PLATFORM(WINCE) && PLATFORM(QT)
    We can not determine the endianess at compile time. For
@@ -634,7 +642,7 @@
 #endif
 
 #if !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32) && !defined(WTF_USE_JSVALUE32_64)
-#if PLATFORM(X86_64) && (PLATFORM(DARWIN) || PLATFORM(LINUX)) || PLATFORM(SPARC64)
+#if PLATFORM(X86_64) && (PLATFORM(DARWIN) || PLATFORM(LINUX)) || PLATFORM(SPARC64) || PLATFORM(S390x)
 #define WTF_USE_JSVALUE64 1
 #elif PLATFORM(ARM) || PLATFORM(PPC64)
 #define WTF_USE_JSVALUE32 1
