Index: trunk/Source/JavaScriptCore/dfg/DFGCapabilities.cpp
===================================================================
--- trunk/Source/JavaScriptCore/dfg/DFGCapabilities.cpp	(revision 98179)
+++ trunk/Source/JavaScriptCore/dfg/DFGCapabilities.cpp	(revision 113389)
@@ -62,4 +62,6 @@
 bool canCompileOpcodes(CodeBlock* codeBlock)
 {
+    if (!MacroAssembler::supportsFloatingPoint())
+        return false;
     return canHandleOpcodes<canCompileOpcode>(codeBlock);
 }
