Index: /trunk/Source/JavaScriptCore/ChangeLog
===================================================================
--- /trunk/Source/JavaScriptCore/ChangeLog	(revision 113365)
+++ /trunk/Source/JavaScriptCore/ChangeLog	(revision 113389)
@@ -1,2 +1,14 @@
+2012-04-05  Oliver Hunt  <oliver@apple.com>
+
+        SIGILL in JavaScriptCore on a Geode processor
+        https://bugs.webkit.org/show_bug.cgi?id=82496
+
+        Reviewed by Gavin Barraclough.
+
+        Don't attempt to use the DFG when SSE2 is not available.
+
+        * dfg/DFGCapabilities.cpp:
+        (JSC::DFG::canCompileOpcodes):
+
 2012-04-05  Oliver Hunt  <oliver@apple.com>
 
Index: /trunk/Source/JavaScriptCore/dfg/DFGCapabilities.cpp
===================================================================
--- /trunk/Source/JavaScriptCore/dfg/DFGCapabilities.cpp	(revision 98179)
+++ /trunk/Source/JavaScriptCore/dfg/DFGCapabilities.cpp	(revision 113389)
@@ -62,4 +62,6 @@
 bool canCompileOpcodes(CodeBlock* codeBlock)
 {
+    if (!MacroAssembler::supportsFloatingPoint())
+        return false;
     return canHandleOpcodes<canCompileOpcode>(codeBlock);
 }
